test/upgrade/installation/serverless.go

```diff
@@ -42,5 +49,51 @@ func UpgradeServerless(ctx *test.Context) error {
 		ctx.T.Error("Eventing upgrade failed:", err)
 	}
 
+	// KnativeKafka doesn't provide the version to wait for. Choosing a well-known image and
+	// waiting for all pods to be updated to this image before proceeding.
+	kafkaWebhookImage, err := test.ImageFromCSV(ctx,
+		test.Flags.CSV,
+		test.OperatorsNamespace,
+		"kafka-webhook")
+	if err != nil {
+		ctx.T.Error(err)
+	}
+
+	if err = WaitForPodsWithImage(ctx,
+		knativeEventing,
+		"app=kafka-webhook",
+		"kafka-webhook",
+		kafkaWebhookImage); err != nil {
+		ctx.T.Error(err)
+	}
```

Comment:
To address the edge-cases mentioned below, should we instead check the deployment to be updated and then rolled out? Rollout can be detected via https://github.com/knative/serving/blob/1b945a0942ef9638c3df9af68a5f91b573ee3aad/test/e2e/autoscale.go#L424-L444. So it'd be

1. Wait for the expected image to appear in the deployment spec
2. Wait for the rollout to finish

Reply:
That's an option too but not sure it's necessary. See my previous comment.